### PR TITLE
[IMP] mail: make m2x user tests use start instead of createWebClient

### PR DESCRIPTION
**Pull request by tsm-odoo**
This pr prepare the ground for the main PR. For now, the `m2x_user_test` was only using two models out of those which are given by the `beforeEach` function. When the real model definitions will be used, we will need all the models to be passed because relational fields are fetched according to the models that are available. The issue with passing all the data to the `createWebClient` method is that the `beforeEach` function also adds user ids used both by the mock server and the tests themselves. Those ids are deleted in the `init` method of the mail's legacy mock server. Using the mail's `start` method with the `hasWebClient` parameter set to true result in an instanciation of the mail's legacy mock server thus the removal of those ids.

task-2767820

### CODE REVIEW

**Comment by alexkuhn** [comment]
`m2x_avatar_user_tests.js` are not discuss tests, so I don't really understand why we are making them dependent on discuss stuffs with start now?

File: addons/mail/static/tests/m2x_avatar_user_tests.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ QUnit.module('mail', {}, function () {
             Many2OneAvatarUser.prototype.partnerIds = {};
 
             Object.assign(this.data, {
-                'foo': {
+                'ir.attachment': {
```
Comment (seb-odoo) [inline addons/mail/static/tests/m2x_avatar_user_tests.js:28]: Fields should be in mock model. Keep foo if you need to create fake fields on attachment instead.

I was suggesting attachment for user_id (which exists but is named `create_uid`), but if it doesn't have a x2many on user_id, forget about it for now.

**Review by seb-odoo — APPROVED** [review]
Looks ok, I let @alexkuhn have final check as you mostly checked it with him.

File: addons/mail/static/tests/m2x_avatar_user_tests.js [hunk 2]
```diff
@@ -58,189 +58,57 @@ QUnit.module('mail', {}, function () {
...
         });
 
-        assert.containsN(form, '.o_field_many2manytags.avatar.o_field_widget .badge', 2,
-            "should have 2 records");
-        assert.strictEqual(form.$('.o_field_many2manytags.avatar.o_field_widget .badge:first img').data('src'), '/web/image/res.users/11/avatar_128',
-            "should have correct avatar image");
-
         await dom.click(form.$('.o_field_many2manytags.avatar .badge:first .o_m2m_avatar'));
         await dom.click(form.$('.o_field_many2manytags.avatar .badge:nth(1) .o_m2m_avatar'));
+        assert.containsN(document.body, '.o_ChatWindow', 2, '2 Chat windows should be opened');
```
Comment (alexkuhn) [inline addons/mail/static/tests/m2x_avatar_user_tests.js:99]: I don't think click on 2 different avatar and have 2 chat windows is relevant for this test. Could just test with a single click => single chat window so that test is much simpler.

File: addons/mail/static/tests/m2x_avatar_user_tests.js [hunk 3]
```diff
@@ -58,189 +58,57 @@ QUnit.module('mail', {}, function () {
...
-            "read res.users 11",
-            "read res.users 23",
-        ]);
-
-        list.destroy();
-    });
-
     QUnit.test('many2many_avatar_user in kanban view', async function (assert) {
-        assert.expect(11);
+        assert.expect(4);
```
Comment (alexkuhn) [inline addons/mail/static/tests/m2x_avatar_user_tests.js:111]: Note that this could do even better and split this test in 2: one to assert there is m2m_avatar field widget, and another for popover assertions.

But let's just ignore since this test is out of scope of discuss. It's already a gift to remove complexity.

File: addons/mail/static/tests/m2x_avatar_user_tests.js [hunk 4]
```diff
@@ -58,189 +58,57 @@ QUnit.module('mail', {}, function () {
...
-
-        assert.strictEqual(list.$('.o_data_cell span').text(), 'MarioLuigiMarioYoshi');
-
-        // sanity check: later on, we'll check that clicking on the avatar doesn't open the record
-        await dom.click(list.$('.o_data_row:first span'));
-
         await dom.click(list.$('.o_data_cell:nth(0) .o_m2o_avatar > img'));
-        await dom.click(list.$('.o_data_cell:nth(1) .o_m2o_avatar > img'));
-        await dom.click(list.$('.o_data_cell:nth(2) .o_m2o_avatar > img'));
+        assert.containsOnce(document.body, '.o_ChatWindow', 'Chat window should be opened');
```
Comment (alexkuhn) [inline addons/mail/static/tests/m2x_avatar_user_tests.js:74]: Can remove this LOC (successive assertions should not be separated by newlines).

Just make a newline when it's another step of test, i.e. another group of user interactions followed by assertion(s). The shape helps get the complexity of test at a quick glance. And also the newline should let you know that you could split the test there.

**Review by alexkuhn — COMMENTED** [review]
Looks good.

TBH I don't understand the tests that have been barely affected (now use `start` instead of `createWebClient`)...
...I mean, their relevance of testing something meaningful _functionally_, not about the reasons for the changes in this PR.
But it's not worth caring when these tests are kept mostly out of discuss code scope.

At least the tests that are in discuss scope (the ones with `hasChatWindow`) are nice.

File: addons/mail/static/tests/m2x_avatar_user_tests.js (hunk 4 above)
Comment (alexkuhn) [reply addons/mail/static/tests/m2x_avatar_user_tests.js:74]: The comment is on empty line 89/74. Github shows more context above.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
